src/avt/Queries/Queries/avtXRayImageQuery.C

```diff
@@ -105,13 +107,31 @@ inline void Add(double result[3], const double v1[3], const double v2[3])
     result[2] = v1[2] + v2[2];
 }
 
+inline void Add3(double result[3], 
+                 const double v1[3], 
+                 const double v2[3], 
+                 const double v3[3])
+{
+    result[0] = v1[0] + v2[0] + v3[0];
+    result[1] = v1[1] + v2[1] + v3[1];
+    result[2] = v1[2] + v2[2] + v3[2];
+}
+
 inline void Scale(double result[3], const double v[3], const double s)
 {
     result[0] = v[0] * s;
     result[1] = v[1] * s;
     result[2] = v[2] * s;
 }
 
+inline void Normalize(double result[3], const double v[3])
```

Comment:
Just a comment, since the other vector functions were already added with a previous commit.
avt/Math/avtVector has all these already defined and could have been used.

Reply:
Ah, good to know. I think I implemented all of these myself since I found that that was the approach used in the X Ray Filter. I think a few of the first vector math functions I implemented I stole directly from there.

How about this: I will add an entry to my list of x ray query deliverables for 3.3.2 so I remember to swap out the vector math functions here with vector math functions from `avt/Math/avtVector`. I am planning to do one more big PR for the x ray image query for 3.3.2 where I refactor code, clean everything up, and simplify things. To keep this PR focused around the ray outputs, how about I'll switch to the vector math functions in `avt/Math/avtVector` in the next PR?